src/cryptonote_core/service_node_list.cpp

```diff
@@ -1892,7 +1892,7 @@ namespace service_nodes
 
   static crypto::hash hash_uptime_proof(const cryptonote::NOTIFY_UPTIME_PROOF::request &proof, uint8_t hf_version)
   {
-    auto buf = tools::memcpy_le(proof.pubkey.data, proof.timestamp, proof.public_ip, proof.storage_port, proof.pubkey_ed25519.data, proof.qnet_port);
+    auto buf = tools::memcpy_le(proof.pubkey.data, proof.timestamp, proof.public_ip, proof.storage_port, proof.storage_lmq_port, proof.pubkey_ed25519.data, proof.qnet_port);
```

Comment:
This is going to break uptime proof compatibility with existing 6.x nodes.  Take a look at what we did here, when adding the ed25519 keys and quorumnet port to the hash:

https://github.com/loki-project/loki-core/blob/c00e8221ccea6872baa7ee96c61203e45dfd7a60/src/cryptonote_core/service_node_list.cpp#L1901-L1911

Basically, put the new field at the end of the data, and then don't actually include it when calculating the hash (by subtracting off `buf_size`, below) until we hit hard fork 15.

Reply:
updated